noir-projects/aztec-nr/value-note/src/utils.nr

```diff
@@ -55,17 +70,24 @@ pub fn decrement_by_at_most(
         change_value = decremented - max_amount;
         decremented -= change_value;
     }
-    increment(balance, change_value, owner);
+    increment(context, balance, change_value, owner);
 
     decremented
 }
 
 // Removes the note from the owner's set of notes.
 // Returns the value of the destroyed note.
-pub fn destroy_note(balance: PrivateSet<ValueNote>, owner: AztecAddress, note: ValueNote) -> Field {
+pub fn destroy_note(
+    mut context: PrivateContext,
+    balance: PrivateSet<ValueNote>,
+    owner: AztecAddress,
+    note: ValueNote
+) -> Field {
     // Ensure the note is actually owned by the owner (to prevent user from generating a valid proof while
     // spending someone else's notes).
-    assert(note.owner.eq(owner));
+    let owner_npk_m_hash = get_npk_m_hash(&mut context, owner);
+
+    assert(note.owner_npk_m_hash.eq(owner_npk_m_hash));
```

Comment:
As mentioned, this is broken and should be fixed in #6312. Should be handled by the application logic, nullifier already handling that only he can nullify it.